pkg/wasp/pod-ranker/rank_utils.go

```diff
@@ -134,7 +134,13 @@ func exceedMemoryLimits(summary summaryFunc) cmpFunc {
 func hasContainerExceedMemoryLimits(pod *v1.Pod, summary stats_collector.PodSummary) bool {
 	for _, container := range pod.Spec.Containers {
 		if rQuantity, ok := container.Resources.Limits[v1.ResourceMemory]; ok {
-			memoryAndSwapSumQuantity := memoryAndSwapUsage(*summary.Containers[container.Name].MemoryWorkingSetBytes, *summary.Containers[container.Name].MemorySwapCurrentBytes)
+			containerStats, exists := summary.Containers[container.Name]
+			if !exists {
+				continue
+			}
+			workingSet := *containerStats.MemoryWorkingSetBytes
+			swapUsage := *containerStats.MemorySwapCurrentBytes
+			memoryAndSwapSumQuantity := memoryAndSwapUsage(workingSet, swapUsage)
```

Comment:
Hey, what do you think about also checking if these are nil? Maybe we can log a message in that case for better debuggability.